### PR TITLE
Update pipeline section in add flows based on new UX

### PR DESCRIPTION
**Pull request by rohitkrai03**
**Fixes**:  https://issues.redhat.com/browse/ODC-3285

**Analysis / Root cause**:  The UX changed a bit for the pipeline section in the add flows.

**Solution Description**:  Address the UX changes based on the new design docs.

**Screen shots / Gifs for design review**: 

Peek 2020-03-31 01-13

**Browser conformance**: 

- [X] Chrome
- [X] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/dev-console/src/components/import/pipeline/PipelineSection.tsx [hunk 1]
```diff
@@ -59,7 +59,7 @@ const PipelineSection: React.FC<PipelineSectionProps> = ({ flags }) => {
           <Alert
             isInline
             variant="info"
-            title="Select a builder image to see if there is a pipeline template available for this runtime."
+            title="Select a builder image and resource to see if there is a pipeline template available for this runtime."
```
Comment (andrewballantyne) [inline frontend/packages/dev-console/src/components/import/pipeline/PipelineSection.tsx]: Need to handle some preview information to help the user. [Designs](https://docs.google.com/document/d/1mrNqDt8SzTafBf-sKfqykKy-cHV4py7WII0rTNcyA3c/edit?disco=AAAAJOTSk6U). 

We should have the information from the form to help build this. If you don't have a builder image, we can show this generic message.

Probably the easiest way to test this is to select a builder image that doesn't have one of the templates... like `Nginx`

Comment (andrewballantyne) [reply frontend/packages/dev-console/src/components/import/pipeline/PipelineSection.tsx]: Ah, sorry, this is not the right Alert. The one in `PipelineTemplate` is the one that needs the advanced wording.

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/import/pipeline/PipelineSection.tsx]: Updated the text to be more specific using builder image and resources value from the form.

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/import/pipeline/PipelineSection.tsx]: But this one here doesn't need to be changed as this is the empty state when no builder image is selected. I've updated this text according to UX.

Comment (andrewballantyne) [reply frontend/packages/dev-console/src/components/import/pipeline/PipelineSection.tsx]: Yup sorry, added the comment and then saw it was the wrong one... wasn't a spot for my to make the other comment so I left it -- was lazy, sorry about that.

File: frontend/packages/dev-console/src/components/import/pipeline/PipelineTemplate.tsx [hunk 2]
```diff
@@ -11,12 +11,17 @@ import { Pipeline } from '../../../utils/pipeline-augment';
...
-const MISSING_RUNTIME_LABEL_TEXT = 'There are no pipeline templates available for this runtime.';
 
 const labelType = 'pipeline.openshift.io/type';
 const labelRuntime = 'pipeline.openshift.io/runtime';
 const labelDocker = 'pipeline.openshift.io/strategy';
 
+const getAlertText = (isDockerStrategy, builderImage, resourceType) => {
+  if (isDockerStrategy) return MISSING_DOCKERFILE_LABEL_TEXT;
+
+  return `There are no pipeline templates available for ${builderImage} and ${resourceType} combination.`;
```
Comment (andrewballantyne) [inline frontend/packages/dev-console/src/components/import/pipeline/PipelineTemplate.tsx]: This is not going to render a really understandable message... at least not one that is formatted to match the form.

I thought we had a map of ResourceType to renderable item, but it seems we copped out when we did the ResourceSection of the form (I was reviewing that, I should have requested it 😞). Oh wells, no sense in crying over spilled milk.

Need two changes:

* A map made next to the `Resources` enum that converts `resourceType` to a readable string
    * Go update `ResourceSection` to use that map (so we stay in sync)
* Thread `builderImages` into `PipelineSection` and down to here to allow you to get the `title` attribute

That should give you a readable string that makes sense with the rest of the form.

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/import/pipeline/PipelineTemplate.tsx]: Done.

File: frontend/packages/dev-console/src/components/import/import-types.ts [hunk 3]
```diff
@@ -3,6 +3,8 @@ import { K8sResourceKind, ContainerPort } from '@console/internal/module/k8s';
 import { LazyLoader } from '@console/plugin-sdk';
 import { NameValuePair, NameValueFromPair } from '@console/shared';
 import { NormalizedBuilderImages } from '../../utils/imagestream-utils';
+import { DeploymentModel, DeploymentConfigModel } from '@console/internal/models';
+import { ServiceModel } from '@console/knative-plugin';
```
Comment (andrewballantyne) [inline frontend/packages/dev-console/src/components/import/import-types.ts]: Order please.

File: frontend/packages/dev-console/src/components/import/pipeline/PipelineTemplate.tsx [hunk 4]
```diff
@@ -8,16 +8,27 @@ import { CLUSTER_PIPELINE_NS } from '../../../const';
...
 const MISSING_DOCKERFILE_LABEL_TEXT =
   'The pipeline template for Dockerfiles is not available at this time.';
-const MISSING_RUNTIME_LABEL_TEXT = 'There are no pipeline templates available for this runtime.';
 
 const labelType = 'pipeline.openshift.io/type';
 const labelRuntime = 'pipeline.openshift.io/runtime';
 const labelDocker = 'pipeline.openshift.io/strategy';
 
-const PipelineTemplate: React.FC = () => {
+const getAlertText = (isDockerStrategy, builderImage, resourceType) => {
```
Comment (andrewballantyne) [inline frontend/packages/dev-console/src/components/import/pipeline/PipelineTemplate.tsx]: Nit, let's type it for sanity sake.
```suggestion
const getAlertText = (isDockerStrategy: boolean, builderImage: string, resourceType: string): string => {
```

**Comment by andrewballantyne** [comment]
During the next fix, could you squash as well? I'll likely be giving a LGTM at that point.

File: frontend/packages/dev-console/src/components/import/pipeline/PipelineTemplate.tsx (hunk 4 above)
Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/import/pipeline/PipelineTemplate.tsx]: Done

File: frontend/packages/dev-console/src/components/import/import-types.ts (hunk 3 above)
Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/import/import-types.ts]: Done

**Pull request merged**